ui/src/services/NodeVolumesUtils.js

```diff
@@ -276,3 +276,28 @@ export const getVolumeListData = createSelector(
     return nodeVolumes.sort((a, b) => compareHealth(b.health, a.health));
   },
 );
+
+export const formatDataForBatchVolumeCreation = (newVolumes) => {
```

Comment:
I would add a simple unit test for the two branches of this function, also as mentioned in the usage of this function I would handle the `!multiVolumeCreation` case here and return a single entry array containing newVolumes, that would allow a simpler usage of the function as its ouput will always be of the same shape `(newVolumes: VolumeCreationFormState) => Volume[]` instead of `(newVolumes: VolumeCreationFormState) => Volume[] | void`